core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/DealiasLCAFilter.scala

```diff
@@ -72,11 +81,7 @@ class DealiasLCAFilter extends Rule[LogicalPlan] with IRHelpers {
       if (alias.isEmpty) {
         item
       } else {
-        val replacement = alias.get._2
-        item transform {
-          case name: Name => name.makeCopy(Array(replacement))
-          case id: Id => id.makeCopy(Array(replacement.asInstanceOf[AnyRef], id.caseSensitive.asInstanceOf[AnyRef]))
-        }
+        alias.get._2
```

Comment:
The whole function could become:
```scala
Option(item).collect{
  case Name(name) => name
  case Id(id, _) => id
}
.flatMap(aliases.get)
.getOrElse(item)
```
or


```scala 
val key = item match {
  case Name(name) => name
  case Id(id, _) => name
  case _ => null
}
aliases.getOrElse(key, item)
```

The former being slightly preferable, even though it is unlikely that `aliases` would have a entry with a `null` key.

Reply:
outdated